pkg/cloud/cloud.go

```diff
@@ -343,6 +343,13 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 	}
 
 	if err := c.waitForVolume(ctx, volumeID); err != nil {
+		// To avoid leaking volume, we should delete the volume just created
+		// TODO: Need to figure out how to handle DeleteDisk failed scenario instead of just log the error
+		if _, error := c.DeleteDisk(ctx, volumeID); error != nil {
```

Comment:
So delete works if the volume is not available when we make the call?

Reply:
Yeah, I did a test on my account. One concern here is  if the deleteVolume call failed as well then we can not prevent the volume get leaked anyway..

Reply:
Sure, that makes sense.